test/integration/Permit2FeeCharging.t.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Test} from 'forge-std/Test.sol';
+import {IERC20} from 'openzeppelin-contracts/contracts/token/ERC20/IERC20.sol';
+import {SafeCast160} from 'permit2/libraries/SafeCast160.sol';
+import {Router} from 'src/Router.sol';
+import {IParam} from 'src/interfaces/IParam.sol';
+import {IAgent} from 'src/interfaces/IAgent.sol';
+import {ERC20Permit2Utils} from 'test/utils/ERC20Permit2Utils.sol';
+import {TypedDataSignature} from 'test/utils/TypedDataSignature.sol';
+
+contract Permit2FeeCalculatorTest is Test, ERC20Permit2Utils, TypedDataSignature {
+    using SafeCast160 for uint256;
+
+    event FeeCharged(address indexed token, uint256 amount, bytes32 metadata);
+
+    bytes4 public constant PERMIT2_TRANSFER_FROM_SELECTOR =
+        bytes4(keccak256(bytes('transferFrom(address,address,uint160,address)')));
+    address public constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address public constant PERMIT2_ADDR = 0x000000000022D473030F116dDEE9F6B43aC78BA3;
+    uint256 public constant BPS_BASE = 10_000;
+    bytes32 public constant META_DATA = bytes32(bytes('permit2:pull-token'));
+
+    address public user;
+    uint256 public userPrivateKey;
+    address public signer;
+    uint256 public signerPrivateKey;
+    address public feeCollector;
+    Router public router;
+    IAgent public userAgent;
+    address public permit2FeeCalculator;
+    IParam.Logic[] public logicsEmpty;
+
+    function setUp() external {
+        (user, userPrivateKey) = makeAddrAndKey('User');
+        (signer, signerPrivateKey) = makeAddrAndKey('Signer');
+        feeCollector = makeAddr('FeeCollector');
+        address pauser = makeAddr('Pauser');
+
+        // Depoly contracts
+        router = new Router(makeAddr('WrappedNative'), PERMIT2_ADDR, address(this), pauser, feeCollector);
+        vm.prank(user);
+        userAgent = IAgent(router.newAgent());
+        router.addSigner(signer);
+
+        // Setup permit2
+        erc20Permit2UtilsSetUp(user, userPrivateKey, address(userAgent));
+        permitToken(IERC20(USDC));
+
+        vm.label(address(router), 'Router');
+        vm.label(address(userAgent), 'UserAgent');
+        vm.label(feeCollector, 'FeeCollector');
+        vm.label(PERMIT2_ADDR, 'Permit2Address');
```

Comment:
`Permit2Address` -> `Permit2`
I think Permit2 is clear enough.

Reply:
Changed.